datasets/medical_dialog/medical_dialog.py

```diff
@@ -0,0 +1,280 @@
+        id_ = -1
+        for filepath in filepaths:
+            with open(filepath, encoding="utf-8") as f_in:
+                # Parameters to just "sectionize" the raw data
+                last_part = ""
+                last_dialog = {}
+                last_list = []
+                last_user = ""
+                check_list = []
+
+                # These flags are present to have a single function address both chinese and english data
+                # English data is a little hahazard (i.e. the sentences spans multiple different lines),
+                # Chinese is compact with one line for doctor and patient.
+                conv_flag = False
+                des_flag = False
+
+                while True:
+                    line = f_in.readline()
+                    if not line:
+                        break
+
+                    # Extracting the dialog id
+                    if line[:2] == "id":  # Hardcode alert!
+                        # Handling ID references that may come in the description
+                        # These were observed in the Chinese dataset and were not
+                        # followed by numbers
+                        try:
+                            dialogue_id = int(re.findall(r"\d+", line)[0])
+                        except IndexError:
+                            continue
+
+                    # Extracting the url
+                    if line[:4] == "http":  # Hardcode alert!
+                        dialogue_url = line.rstrip()
+
+                    # Extracting the patient info from description.
+                    if line[:11] == "Description":  # Hardcode alert!
+                        last_part = "description"
+                        last_dialog = {}
+                        last_list = []
+                        last_user = ""
+                        last_conv = {"speaker": "", "utterance": ""}
+                        while True:
+                            line = f_in.readline()
+                            if (not line) or (line in ["\n", "\n\r"]):
+                                break
+                            else:
+                                if data_lang == "zh":  # Condition in chinese
+                                    if line[:5] == "病情描述：":  # Hardcode alert!
+                                        last_user = "病人"
+                                        sen = f_in.readline().rstrip()
+                                        des_flag = True
+
+                                if data_lang == "en":
+                                    last_user = "Patient"
+                                    sen = line.rstrip()
+                                    des_flag = True
+
+                                if des_flag:
+                                    if sen == "":
+                                        continue
+                                    if sen[-1] not in ".。？，,?!！~～":
+                                        sen += "."
```

Comment:
same here

Reply:
I removed the two pre-processing lines but now the circleci tests are failing.